STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Validations.cs

```diff
@@ -0,0 +1,204 @@
+                (Rule: IsInvalid(httpExchange), Parameter: nameof(HttpExchange)),
+                (Rule: IsInvalid(defaultHttpMethod), Parameter: nameof(HttpMethod)),
+                (Rule: IsInvalid(defaultHttpVersion), Parameter: nameof(Version)),
+                (Rule: IsInvalid(defaultHttpVersionPolicy), Parameter: nameof(HttpVersionPolicy)));
+
+            ValidateHttpExchange(
+                httpExchange,
+                defaultHttpMethod,
+                defaultHttpVersion,
+                defaultHttpVersionPolicy);
+        }
+
+        private static void ValidateHttpExchange(
+            HttpExchange httpExchange,
+            HttpMethod httpMethod,
+            Version httpVersion,
+            HttpVersionPolicy httpVersionPolicy)
+        {
+            ValidateHttpExchangeRequest(
+                httpExchangeRequest: httpExchange.Request,
+                httpMethod,
+                httpVersion,
+                httpVersionPolicy);
+        }
+
+        private static void ValidateHttpExchangeRequest(
+            HttpExchangeRequest httpExchangeRequest,
+            HttpMethod httpMethod,
+            Version httpVersion,
+            HttpVersionPolicy httpVersionPolicy)
+        {
+            ValidateHttpExchangeRequestNotNull(httpExchangeRequest);
+
+            ValidateInputArgument(
+                (Rule: IsInvalid(httpExchangeRequest.BaseAddress),
+                    Parameter: nameof(HttpExchangeRequest.BaseAddress)),
+
+                (Rule: IsInvalid(httpExchangeRequest.RelativeUrl),
+                    Parameter: nameof(HttpExchangeRequest.RelativeUrl)),
+
+                (Rule: IsInvalidHttpMethod(httpExchangeRequest.HttpMethod, httpMethod),
+                    Parameter: nameof(HttpExchangeRequest.HttpMethod)),
+
+                (Rule: IsInvalidHttpVersion(httpExchangeRequest.Version, httpVersion),
+                    Parameter: nameof(HttpExchangeRequest.Version)),
+
+                (Rule: IsInvalidHttpVersionPolicy(httpExchangeRequest.VersionPolicy, httpVersionPolicy),
+                    Parameter: nameof(HttpExchangeRequest.VersionPolicy)));
+
+            ValidateHttpExchangeRequestHeaders(httpExchangeRequestHeaders: httpExchangeRequest.Headers);
+        }
+
+        private static void ValidateHttpMethod(HttpExchangeRequest httpExchangeRequest, HttpMethod httpMethod)
+        {
+            ValidateInputArgument((
+                Rule: IsInvalidHttpMethod(httpExchangeRequest.HttpMethod, httpMethod),
+                Parameter: nameof(HttpExchangeRequest.HttpMethod)));
+        }
+
+        private static void ValidateVersion(HttpExchangeRequest httpExchangeRequest, Version httpVersion)
+        {
+            ValidateInputArgument((
+                Rule: IsInvalidHttpVersion(httpExchangeRequest.Version, httpVersion),
+                Parameter: nameof(HttpExchangeRequest.Version)));
+        }
+
+        private static void ValidateHttpExchangeNotNull(HttpExchange httpExchange)
```

Comment:
This method is not used and can be deleted.  Do a check everywhere to remove unused code as it is not easy to spot on Github review ide